irods/resource.py

```diff
@@ -38,6 +41,54 @@ def __init__(self, manager, result=None):
 
         self._meta = None
 
+
+    ## Cached properties to expose parent id or name regardless whether the DB model is iRODS 4.1- or 4.2+
+
+    @property
+    def parent_id(self):
+        if self.parent is None:
+            return None
+        if self._parent_id == '':
+            sess = self.manager.sess
+            if sess.server_version >= (4, 2, 0):
+                self._parent_id = self.parent
+            else:
+                self._parent_id = sess.query(Resource).filter(Resource.name == self.parent).one()[Resource.id]
+        return int(self._parent_id)
+
+
+    @property
+    def parent_name(self):
+        if self.parent is None:
+            return None
+        if self._parent_name == '':
+            sess = self.manager.sess
+            if sess.server_version < (4, 2, 0):
+                self._parent_name = self.parent
+            else:
+                self._parent_name = sess.query(Resource).filter(Resource.id == self.parent).one()[Resource.name]
+        return self._parent_name
+
+    ## Cached property to expose resource hierarchy string
+
+    @property
+    def hierarchy(self):
+        if self._hierarchy == '':
+            self._hierarchy = ';'.join(r.name for r in reversed(self.get_parent_chain()))
+        return self._hierarchy
+
+    ## Retrieve chain of parent objects to top level parent
+
+    def get_parent_chain(self):
```

Comment:
There is a function in `server/core` called `irods::leaf_id_to_hier` which effectively does the same thing as this function. I personally find that parlance easier to understand than "parent chain". Should we consider a name similar to the other function in the server? Just trying to bring together a common language, I guess...

Reply:
In this case, it's not necessarily a leaf, but we could do something in that spirit.  How about `node_to_hier`, then?

Reply:
or `node_to_topdown_hier` perhaps? not sure what folks would prefer.